### PR TITLE
Getter for the ARZ Model to query shower profiles

### PR DESCRIPTION
**Pull request by colemanalan**
Issue being addressed:

There isn't a way to directly extract the charge-excess profile distributions in the ARZ model. This PR simply adds a getter to query them directly from the ARZ class

### CODE REVIEW

File: NuRadioMC/SignalGen/ARZ/ARZ.py [hunk 1]
```diff
@@ -483,10 +515,6 @@ def get_time_trace(self, shower_energy, theta, N, dt, shower_type, n_index, R, s
             number of samples in the time domain
         dt: float
             size of one time bin in units of time
-        profile_depth: array of floats
```
Comment (colemanalan) [inline NuRadioMC/SignalGen/ARZ/ARZ.py:486]: This PR also fixes these doc-strings which refer to arguments that do not exist.

File: NuRadioMC/SignalGen/ARZ/ARZ.py [hunk 2]
```diff
@@ -468,6 +468,38 @@ def set_interpolation_factor2(self, interp_factor):
...
+    def get_shower_profile(self, shower_energy, shower_type, iN):
+        """
+        Gets a charge-excess profile
+
+        Parameters
+        ----------
+        shower_energy: float
+            the energy of the shower
+        shower_type: string (default "HAD")
+            type of shower, either "HAD" (hadronic), "EM" (electromagnetic) or "TAU" (tau lepton induced)
```
Comment (fschlueter) [inline NuRadioMC/SignalGen/ARZ/ARZ.py:480]: Do we still store "TAU" shower profiles? I thought tau showers are handled differently nowadays

File: NuRadioMC/SignalGen/ARZ/ARZ.py [hunk 3]
```diff
@@ -468,6 +468,38 @@ def set_interpolation_factor2(self, interp_factor):
...
+        iN: int
+            specify shower number
+
+        Returns
+        -------
+        efield_trace: two arrays of floats
```
Comment (fschlueter) [inline NuRadioMC/SignalGen/ARZ/ARZ.py:486]: "efield_trace" ?

File: NuRadioMC/SignalGen/ARZ/ARZ.py [hunk 4]
```diff
@@ -468,6 +468,38 @@ def set_interpolation_factor2(self, interp_factor):
...
+            slant depths and charge profile amplitudes
+        """
+
+        energies = np.array([*self._library[shower_type]])
+        iE = np.argmin(np.abs(energies - shower_energy))
+
+        rescaling_factor = shower_energy / energies[iE]
+
+        profiles = self._library[shower_type][energies[iE]]
+        N_profiles = len(profiles['charge_excess'])
```
Comment (fschlueter) [inline NuRadioMC/SignalGen/ARZ/ARZ.py:496]: variables should start lower case right?

Comment (fschlueter) [reply NuRadioMC/SignalGen/ARZ/ARZ.py:496]: I think the variable is also not used

File: NuRadioMC/SignalGen/ARZ/ARZ.py (hunk 2 above)
Comment (colemanalan) [reply NuRadioMC/SignalGen/ARZ/ARZ.py:480]: This is just to be symmetric with `get_time_trace`

Comment (fschlueter) [reply NuRadioMC/SignalGen/ARZ/ARZ.py:480]: But there is no keyword "TAU" in the shower library, or is there?

Comment (colemanalan) [reply NuRadioMC/SignalGen/ARZ/ARZ.py:480]: Yes they exist.

Comment (colemanalan) [reply NuRadioMC/SignalGen/ARZ/ARZ.py:480]: Ah I see what you mean.
I checked and they do not. I can pull this out in all docstrings

**Review by fschlueter — APPROVED** [review]

**Comment by fschlueter** [comment]
@colemanalan you can go ahead and merge this

**Pull request merged**